src/main/java/org/jabref/logic/exporter/CffExporter.java

```diff
@@ -0,0 +1,227 @@
+            Map<String, Object> preferredData = new LinkedHashMap<>();
+            Map<String, Object> data = originalData;
+            data.put("cff-version", "1.2.0");
+
+            for (BibEntry entry : entries) {
+                // Retrieve all fields
+                Map<Field, String> entryMap = entry.getFieldMap();
+
+                // Compulsory message field
+                String message = entryMap.getOrDefault(StandardField.COMMENT,
+                        "If you use this software, please cite it using the metadata from this file.");
+                data.put("message", message);
+                entryMap.remove(StandardField.COMMENT);
+
+                // Compulsory title field
+                String title = entryMap.getOrDefault(StandardField.TITLE, "No title specified.");
+                data.put("title", title);
+                entryMap.remove(StandardField.TITLE);
+
+                // Compulsory authors field
+                List<Author> authors = AuthorList.parse(entryMap.getOrDefault(StandardField.AUTHOR, ""))
+                                                 .getAuthors();
+
+                // Create two copies of the same list to avoid using YAML anchors and aliases
+                List<Map<String, String>> authorsList = new ArrayList<>();
+                List<Map<String, String>> authorsListPreferred = new ArrayList<>();
+                authors.forEach(author -> {
+                    Map<String, String> authorMap = new LinkedHashMap<>();
+                    Map<String, String> authorMapPreferred = new LinkedHashMap<>();
+                    if (author.getFamilyName().isPresent()) {
+                        authorMap.put("family-names", author.getFamilyName().get());
+                        authorMapPreferred.put("family-names", author.getFamilyName().get());
+                    }
+                    if (author.getGivenName().isPresent()) {
+                        authorMap.put("given-names", author.getGivenName().get());
+                        authorMapPreferred.put("given-names", author.getGivenName().get());
+                    }
+                    if (author.getNamePrefix().isPresent()) {
+                        authorMap.put("name-particle", author.getNamePrefix().get());
+                        authorMapPreferred.put("name-particle", author.getNamePrefix().get());
+                    }
+                    if (author.getNameSuffix().isPresent()) {
+                        authorMap.put("name-suffix", author.getNameSuffix().get());
+                        authorMapPreferred.put("name-suffix", author.getNameSuffix().get());
+                    }
+                    authorsList.add(authorMap);
+                    authorsListPreferred.add(authorMapPreferred);
+                });
+                data.put("authors", authorsList.isEmpty() ? "No author specified." : authorsList);
+                entryMap.remove(StandardField.AUTHOR);
+
+                // Type
```

Comment:
Extract each block to a separate method. Then, the code is more structured and readable.

Reply:
I added a method `parseAuthors` and a method `parseDate`. For the type, it would not be logical to add a specific method since it switches the variable `data` from `originalData` to `referencesData`. This has to belong to the main body of the function.